### PR TITLE
Add -auxfromfile option to IJ driver

### PR DESCRIPTION
**Pull request by victorapm**
Add `-auxfromfile` option for reading an auxiliary matrix from file, which is then used to build the preconditioner. 

This is useful, for example, for the case that a filtered version of `A` is used to build the preconditioner.

### CODE REVIEW

**Comment by liruipeng** [comment]
So, the purpose is to build preconditioner from a matrix M and use it to precondition the original Ax=b? What is the application for this? Thanks!

**Comment by victorapm** [comment]
@liruipeng, correct. One example of application is testing the separate displacement component strategy that geosx uses for solving linear elasticity.

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

**Review by oseikuffuor1 — APPROVED** [review]
Looks good. Thanks Victor!

File: src/test/ij.c [hunk 1]
```diff
@@ -4148,11 +4220,11 @@ main( hypre_int argc,
...
       HYPRE_ParaSailsSetLogging(pcg_precond, ioutdat);
 
-      HYPRE_IJMatrixGetObject( ij_A, &object);
-      parcsr_mat = (HYPRE_ParCSRMatrix) object;
+      HYPRE_IJMatrixGetObject(ij_A, &object);
+      parcsr_A = (HYPRE_ParCSRMatrix) object;
 
-      HYPRE_ParaSailsSetup(pcg_precond, parcsr_mat, NULL, NULL);
-      HYPRE_ParaSailsBuildIJMatrix(pcg_precond, &ij_M);
+      HYPRE_ParaSailsSetup(pcg_precond, parcsr_A, NULL, NULL);
```
Comment (liruipeng) [inline src/test/ij.c]: why it is not set up with `parcsr_M` here? It seems that `parcsr_M` is used in other setups

Comment (victorapm) [reply src/test/ij.c]: I missed that! Thanks for catching!

Comment (liruipeng) [reply src/test/ij.c]: so basically, from now on we should set up a preconditioner with `M` in the `ij` driver? If so, we should give all a heads-up

Comment (victorapm) [reply src/test/ij.c]: Yes, good idea! I'll mention that on Teams

**Review by liruipeng — APPROVED** [review]
I am OK with these changes. We should keep in mind there is another matrix `M` to deal with and in most of the cases `M=A`.

**Comment by liruipeng** [comment]
Please hold on the merge until I give green light. thanks!

**Comment by liruipeng** [comment]
@victorapm You can merge in this PR now. (you have to deal with the conflicts. sorry)

**Pull request merged**